crates/smartmodule-development-kit/src/test.rs

```diff
@@ -75,20 +81,43 @@ impl TestOpt {
 
         let mut chain = chain_builder.initialize(&engine)?;
 
-        // get raw json in one of other ways
-        let raw_input = if let Some(input) = self.text {
+        let mut input_size = 0;
+        let test_record: Vec<Record> = if let Some(input) = self.text {
             debug!(input, "input string");
-            input.as_bytes().to_vec()
-        } else if let Some(json_file) = &self.file {
-            std::fs::read(json_file)?
+            input.as_bytes().to_vec();
+
+            let input_bytes = input.as_bytes().to_vec();
+
+            input_size += input_bytes.len();
+
+            vec![Record::new_key_value(RecordKey::NULL, input_bytes)]
+        } else if let Some(test_file) = &self.file {
+            if self.file_single {
```

Comment:
can we have unit test for this?  Also, we seems to have similar fn in the consume as well.  It would be good to consolidate into a common util (not show stopper).